scripts/zbtestcrypt

```diff
@@ -13,36 +13,72 @@ from scapy.all import *
 from killerbee import *
 from killerbee.scapy_extensions import *
 
+from rangeparser import RangeParser
+
+# parser for packet ranges
+rangeparser= RangeParser()
+
+# encrypt/decrypt packet with appropriate key
+def crypt(packet, data= None, encrypt= False):
+    global link_key
+    global nwk_key
+
+    if packet[ZigbeeSecurityHeader].key_type == 0:
+        if link_key == None:
+            return False, 'FAILED: No LINK_KEY provided'
+        key= link_key
+    elif packet[ZigbeeSecurityHeader].key_type == 1:
+        if nwk_key == None:
+            return False, 'FAILED: No NWK_KEY provided'
+        key= nwk_key
+    else:
+        return False, 'Unknown KEY type'
+
+    if encrypt:
+        return True, kbencrypt(packet, data, key, verbose= args.verbose)
+    else:
+        return True, kbdecrypt(packet, key, verbose= args.verbose)
+
 if __name__ == '__main__':
     # Command-line arguments
     parser = argparse.ArgumentParser(description="zbtestcrypt: \
         Decrypt, re-encrypt and optionally transmit packets for testing and tool development.")
```

Comment:
I'm thinking longer term of moving this, along with some test PCAPs, to create a unittest that runs and remove it as a script. Do you have any thoughts as to if this is needed as a script vs a unittest?

Reply:
I think we should do both. The script is proving very useful as an analysis tool as well so I'm inclined to keep it (and even move it into the 'tools' section?). I will start crearting sample pcaps with specific features.